pkg/querier/querier.go

```diff
@@ -38,9 +38,8 @@ import (
 
 // Config contains the configuration require to create a querier
 type Config struct {
-	Iterators            bool          `yaml:"iterators" category:"advanced"`
-	BatchIterators       bool          `yaml:"batch_iterators" category:"advanced"`
-	QueryIngestersWithin time.Duration `yaml:"query_ingesters_within" category:"advanced"`
```

Comment:
Unfortunately we need to preserve this YAML field for another two Mimir releases, before removing it.

We can hide it by using `doc:"hidden"` tag. If we find that it has non-default value, we can use it to set default limits (in `initRuntimeConfig` in `modules.go`).

Reply:
Updated in https://github.com/grafana/mimir/pull/4287/commits/3edc8444fae6f46c6e56a97ee8cb7097f7975596, is this what you're expecting? I had to do something a little ugly in `RegisterFlags()` to make sure the default value was set before the yaml is parsed.